ADAL/tests/unit/ADAuthenticationErrorTests.m

```diff
@@ -40,70 +40,88 @@ - (void)tearDown
     [super tearDown];
 }
 
-- (void)testNew
+#pragma mark - Initialization
+
+- (void)testNew_shouldThrow
 {
     XCTAssertThrows([ADAuthenticationError new], @"The new selector should not work due to requirement to use the parameterless init. At: '%s'", __PRETTY_FUNCTION__);
 }
 
-- (void)testParameterlessInit
+- (void)testParameterlessInit_shouldThrow
 {
-    XCTAssertThrows([ADAuthenticationError init], @"Parameterless init should throw. At: '%s'", __PRETTY_FUNCTION__);
```

Comment:
Not sure why compiler let it pass. Seems there is +init method in NSObject, but I can not find it in public headers and if it is private, complier should tell us that there is no such method. Will ask on stackoveflow, it is interesting question :). BTW, it throws "NSInvalidArgumentException" when you try to invoke [NSObject init], so it is why the test didn't fail before.